src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs

```diff
@@ -134,9 +134,9 @@ private async Task PollingLoop(CancellationToken cancellationToken)
                 {
                     var result = await GetKvPairs(true, cancellationToken).ConfigureAwait(false);
 
-                    if (result.HasValue() && result.LastIndex > _lastIndex)
+                    if (result.LastIndex > _lastIndex)
                     {
-                        SetData(result);
+                        Data = result.HasValue() ? result.ToConfigDictionary(_source.ConvertConsulKVPairToConfig) : null;
```

Comment:
So yes I think inlining this is better, nice idea. I think we just need to make one more change here now to make this correct. I think it should be:

```csharp
Data = result.HasValue()
    ? result.ToConfigDictionary(_source.ConvertConsulKVPairToConfig)
    : _source.Optional
        ? null
        : Data;
```

Reply:
Ah right, that is the source of the confusion then. I was under the impression that old data should remain if it is not optional and is missing on a reload.

Reply:
One final observation from me. In `FileConfigurationProvider` it sets `Data` to an empty dictionary when the value is missing. We should probably do the same here.

Reply:
Another difference is that FIleConfigurationProvider uses the OrdinalIgnoreCase comparer:
`new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)`
Should we do the same?

Reply:
Yes. We do actually create the dictionary with `StringComparer.OrdinalIgnoreCase` in the [result.ToConfigDictionary](https://github.com/wintoncode/Winton.Extensions.Configuration.Consul/blob/d9705de56f83572fc96d5e880ae24301da92cb18/src/Winton.Extensions.Configuration.Consul/Extensions/KVPairQueryResultExtensions.cs#L30) extension method. So I don't think it matters too much what we do in the empty case as I don't the comparer will be used, but good to be consistent.

Reply:
In fact, if you call `result.ToConfigDictionary` with a `result` that doesn't have any data then it should already do the right thing and create an empty dictionary.

Reply:
Which means we can just write 

```csharp
Data = result.ToConfigDictionary(_source.ConvertConsulKVPairToConfig);
```